src/helpers/import-helper.js

```diff
@@ -12,6 +12,19 @@ async function supportsDynamicImport() {
   }
 }
 
+function isPackageInstalled(packageName) {
+  try {
+    // Try to require the package
+    require.resolve(packageName);
+    return true;
+  } catch (error) {
+    // If require.resolve throws an error, the package is not installed
+    return false;
+  }
+}
+
+const isTypescriptProject = isPackageInstalled('typescript');
```

Comment:
what about having a way to decide it from the CLI? e.g. `npx sequelize-cli migration:generate --name XXX --typescript`

Reply:
This is also how we are going to do it in the new CLI; https://github.com/sequelize/sequelize/blob/main/packages/cli/src/commands/migration/generate.ts